src/systems/filecoin_mining/storage_mining/storage_miner_actor_code.go

```diff
@@ -99,31 +99,62 @@ func (a *StorageMinerActorCode_I) _slashDealsFromFaultReport(rt Runtime, sectorN
 
 	Release(rt, h, st)
 
-	// TODO: Send(StorageMarketActor, ProcessDealSlash)
+	processDealSlashParam := make([]actor.MethodParam, 2)
+	rt.SendCatchingErrors(&msg.InvocInput_I{
+		To_:     addr.StorageMarketActorAddr,
+		Method_: market.MethodProcessDealSlash,
+		Params_: processDealSlashParam,
+	})
+
 }
 
-// construct FaultReport
+func (a *StorageMinerActorCode_I) _slashPledgeForStorageFault(rt Runtime, sectorNumbers []sector.SectorNumber, faultType sector.StorageFaultType) {
+	h, st := a.State(rt)
+
+	affectedPower := block.StoragePower(0)
+	for _, sectorNo := range sectorNumbers {
+
+		utilizationInfo := st._getUtilizationInfo(rt, sectorNo)
+		affectedPower += utilizationInfo.CurrUtilization()
+
+	}
+
+	Release(rt, h, st)
+
+	panic(affectedPower)
```

Comment:
Don't panic

Reply:
Douglas Adams approved. However, I think this particular panic might be there as a TODO, to remind us to serialize the params into `slashPledgeParams` (@zixuanzh is that correct?). If so, may be helpful to consult `init_actor.go` for relevant examples of current serialization/deserialization APIs.

Reply:
yes, will do.